pkg/controller/workload/bpfcache/endpoint.go

```diff
@@ -61,3 +61,21 @@ func (c *Cache) EndpointIterFindKey(workloadUid uint32) []EndpointKey {
 
 	return res
 }
+
+func (c *Cache) GetServiceEndpointKey(serviceIds map[uint32]string, workloadUid uint32) []EndpointKey {
+	var (
+		key   = EndpointKey{}
+		value = EndpointValue{}
+		iter  = c.bpfMap.KmeshEndpoint.Iterate()
+	)
+
+	res := make([]EndpointKey, 0)
+	for iter.Next(&key, &value) {
+		if value.BackendUid == workloadUid {
+			if _, ok := serviceIds[key.ServiceId]; ok {
```

Comment:
I am concerned about the perfoemance.

We can have 100,000 ebndpoints, i think this can be expensive iterating all them

Reply:
fixed